src/model/StringConversionHandler.h

```diff
@@ -65,77 +64,77 @@ struct StringConversionHandler<QVariant>
     }
 };
 
-QString qObjectToString(const QObject *o);
+QString qObjectToString(const QObject *object);
 
-template<typename Type>
-struct StringConversionHandler<Type *, typename std::enable_if<std::is_base_of<InterfaceBase, Type>::value>::type>
+template<typename T>
+struct StringConversionHandler<T*, std::enable_if_t<std::is_base_of<InterfaceBase, T>::value>>
 {
-    static QString toString(const Type *o)
+    static QString toString(const T *object)
     {
-        return qObjectToString(o);
+        return qObjectToString(object);
     }
 };
 
 
-template<typename Type>
-struct StringConversionHandler<Type, typename std::enable_if<std::is_base_of<StructureBase, Type>::value>::type>
+template<typename T>
+struct StringConversionHandler<T, std::enable_if_t<std::is_base_of<StructureBase, T>::value>>
 {
-    static QString toString(const Type &v)
+    static QString toString(const T &value)
     {
-        return v.toString();
+        return value.toString();
     }
 };
 
-
-template<typename Type>
-struct StringConversionHandler<Type, typename std::enable_if<std::is_enum<Type>::value>::type>
+template<typename T>
+struct StringConversionHandler<T, std::enable_if_t<QtPrivate::IsQEnumHelper<T>::Value, T>>
 {
-    static QString toString(const Type &v)
+    static QString toString(T value)
     {
-        return facelift::enumToString(v);
+        return Enum::toString(value);        
     }
 };
 
-template<typename ElementType>
-struct StringConversionHandler<QList<ElementType> >
+template<typename T>
+struct StringConversionHandler<QList<T> >
 {
-    static QString toString(const QList<ElementType> &v)
+    static QString toString(const QList<T> &v)
     {
         QString s;
         QTextStream str(&s);
         str << "[ ";
         for (const auto &element : v) {
-            str << StringConversionHandler<ElementType>::toString(element);
+            str << StringConversionHandler<T>::toString(element);
             str << ", ";
         }
         str << "]";
         return s;
     }
 };
 
-template<typename ElementType>
```

Comment:
Why are you changing those names ??

Reply:
I don't know the reason, why changed names, but I like the new version more. It follows general practices in software for templates, so it should be easier to read for experienced user.

Reply:
yes it's not related to the current changes, but
 typename ElementType, typename Type, typename TypeName are bad in most cases and only T enough here and I just change this small change here.
We need a separate task for replacing  non common template parameters names and
hold something like ElementType only for rare cases when it's needed for better readability.